### PR TITLE
chore(docs): website UI updates

### PR DESCRIPTION
**Pull request by jw-foss**
- Add copy feature for examples
- Remove redundant vp-demo.vue
- Add description section for examples
- Code refactor

Screenshots:

### Description for example and the copy button
image

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/.vitepress/vitepress/components/vp-demo.vue [hunk 1]
```diff
@@ -48,7 +57,23 @@ const props = defineProps({
...
+  },
+  description: {
+    type: String,
+  },
+})
+
+const vm = getCurrentInstance()
+
+const { copy, copied, isSupported } = useClipboard({
+  source: ref(decodeURIComponent(props.rawSource)),
```
Comment (sxzz) [inline docs/.vitepress/vitepress/components/vp-demo.vue:72]: set `read` to `false`

We don't need to read user's clipboard.

File: docs/.vitepress/vitepress/utils.ts [hunk 2]
```diff
@@ -50,15 +50,9 @@ export function createGitHubUrl(
...
-    `/edit` +
-    `/${docsBranch}/` +
-    (docsDir ? docsDir.replace(endingSlashRE, '') + '/' : '') +
-    `${folder ? folder : ''}` +
-    path +
-    `${ext ? ext : ''}`
-  )
+  return `${base.replace(endingSlashRE, '')}/edit/${docsBranch}/${
+    docsDir ? `${docsDir.replace(endingSlashRE, '')}/` : ''
+  }${folder ? folder : ''}${path}${ext ? ext : ''}`
```
Comment (sxzz) [inline docs/.vitepress/vitepress/utils.ts:55]: Why not use `${folder || ''}`

File: docs/.vitepress/vitepress/components/vp-demo.vue (hunk 1 above)
Comment (sxzz) [reply docs/.vitepress/vitepress/components/vp-demo.vue:72]: https://vueuse.org/core/useClipboard/

Comment (JeremyWuuuuu) [reply docs/.vitepress/vitepress/components/vp-demo.vue]: Done!

File: docs/.vitepress/vitepress/utils.ts (hunk 2 above)
Comment (JeremyWuuuuu) [reply docs/.vitepress/vitepress/utils.ts]: Done!

**Comment by JeremyWuuuuu** [comment]
All comments addressed PTAL @sxzz

**Review by sxzz — APPROVED** [review]

**Pull request merged**